src/business-logic/services/rundown-timeline-service.ts

```diff
@@ -232,15 +220,17 @@ export class RundownTimelineService implements RundownService {
     await this.saveRundown(rundown)
   }
 
-  public async insertPieceAsOnAir(rundownId: string, piece: Piece): Promise<void> {
+  public async insertPieceAsOnAir(rundownId: string, piece: Piece, layersToStopPiecesOn: string[] = []): Promise<void> {
     const rundown: Rundown = await this.rundownRepository.getRundown(rundownId)
+
+    rundown.stopActivePiecesOnLayers(layersToStopPiecesOn)
```

Comment:
If `stopActivePiecesOnLayers` returns whether the infinite pieces was affected, we are able to only call `RundownEventEmitter.emitInfinitePiecesUpdatedEvent` when there is an actual change.

Reply:
That is a fine solution. My thoughts were that the result would be something like:
```typescript
interface StoppedPiecesResult {
  infinitePieceIds: string[]
  pieceIdsFromOnAirPart: string[]
}
```
It allows for more granular event emitting.